test/fail_compilation/issue20123.d

```diff
@@ -0,0 +1,44 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/issue20123.d(39): Error: can only `@disable` opUnaryRight
+fail_compilation/issue20123.d(39): Error: can only `@disable` opUnaryRight
+fail_compilation/issue20123.d(40): Error: cannot be used because it is annotated with `@disable`
+fail_compilation/issue20123.d(40): Error: cannot be used because it is annotated with `@disable`
```

Comment:
*WHAT*  cannot be used? 
Also maybe add an `errorSupplemental` pointing to the disabled operator overload?

Reply:
Oof. That needs work.